### PR TITLE
oniguruma: set CMake flags for static builds

### PR DESCRIPTION
**Pull request by gustavderdrache**
###### Motivation for this change

This change makes it possible to build oniguruma statically, based on what I found in the [CMakeLists.txt](https://github.com/kkos/oniguruma/blob/master/CMakeLists.txt#L9). Ultimately this unblocks my building PHP 7.4 statically (since multibyte support requires oniguruma now).

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/top-level/static.nix [hunk 1]
```diff
@@ -243,4 +243,6 @@ in {
   ) super.ocaml-ng;
   
   python27 = super.python27.override { static = true; };
+
+  oniguruma = super.oniguruma.overrideAttrs (_: { cmakeFlags = ["-DBUILD_SHARED_LIBS=OFF"]; });
```
Comment (grahamc) [inline pkgs/top-level/static.nix]: I think it is more standard for these sort of per-package overrides to be documented in the package  definition itself. That said, I'm not certain why these other ones are in here :)

Comment (gustavderdrache) [reply pkgs/top-level/static.nix]: Oops, I thought this was the convention. What's a good model for me to refer to?

Comment (tobim) [reply pkgs/top-level/static.nix]: Typically you add an explicit `static` (or `staticOnly`) parameter to the package expression. Take a look at brotli, you should be able to copy the flags handling verbatim.

**Review by tobim — APPROVED** [review]
Building `pkgsStatic.jq` works thanks to this.

**Review by tobim — CHANGES_REQUESTED** [review]
I didn't realize this would introduce a lot of rebuilds for the non-static variant too. This PR should target staging now.

**Comment by gustavderdrache** [comment]
I've switched to targeting staging, so this should be ready for a re-review.

**Review by tobim — APPROVED** [review]
This looks fine from my pov. Maybe someone with commit bits can do a final check?
cc @matthewbauer @nh2

**Comment by knedlsepp** [comment]
Related: https://github.com/NixOS/nixpkgs/pull/76659

**Comment by knedlsepp** [comment]
Since #76659 was merged (to staging), I don't think this is needed anymore.

**Pull request closed without merging**